media-proxy/src/mesh/st2110.cc

```diff
@@ -0,0 +1,182 @@
+        st_fmt = ST_FRAME_FMT_YUV420CUSTOM8;
+        break;
+    case MESH_VIDEO_PIXEL_FORMAT_YUV422P:
+        st_fmt = ST_FRAME_FMT_YUV422PLANAR8;
+        break;
+    case MESH_VIDEO_PIXEL_FORMAT_YUV422P10LE:
+        st_fmt = ST_FRAME_FMT_YUV422PLANAR10LE;
+        break;
+    case MESH_VIDEO_PIXEL_FORMAT_YUV444P10LE:
+        st_fmt = ST_FRAME_FMT_YUV444PLANAR10LE;
+        break;
+    case MESH_VIDEO_PIXEL_FORMAT_RGB8:
+        st_fmt = ST_FRAME_FMT_RGB8;
+        break;
+    default:
+        return -1; // Error: unknown format
+    }
+
+    return 0; // Success
+}
+
+int ST2110::mesh_audio_format_to_st_format(int mesh_fmt, st30_fmt& st_fmt) {
+    switch (mesh_fmt) {
+    case MESH_AUDIO_FORMAT_PCM_S8:
+        st_fmt = ST30_FMT_PCM8;
+        break;
+    case MESH_AUDIO_FORMAT_PCM_S16BE:
+        st_fmt = ST30_FMT_PCM16;
+        break;
+    case MESH_AUDIO_FORMAT_PCM_S24BE:
+        st_fmt = ST30_FMT_PCM24;
+        break;
+    default:
+        return -1; // Error: unknown format
+    }
+
+    return 0; // Success
+}
+
+int ST2110::mesh_audio_sampling_to_st_sampling(int sampling, st30_sampling& st_sampling) {
+    switch (sampling) {
+    case MESH_AUDIO_SAMPLE_RATE_48000:
+        st_sampling = ST30_SAMPLING_48K;
+        break;
+    case MESH_AUDIO_SAMPLE_RATE_96000:
+        st_sampling = ST30_SAMPLING_96K;
+        break;
+    case MESH_AUDIO_SAMPLE_RATE_44100:
+        st_sampling = ST31_SAMPLING_44K;
+        break;
+    default:
+        return -1; // Error: unknown sampling rate
+    }
+
+    return 0; // Success
+}
+
+int ST2110::mesh_audio_ptime_to_st_ptime(int ptime, st30_ptime& st_ptime) {
+    switch (ptime) {
+    case MESH_AUDIO_PACKET_TIME_1MS:
+        st_ptime = ST30_PTIME_1MS;
+        break;
+    case MESH_AUDIO_PACKET_TIME_125US:
+        st_ptime = ST30_PTIME_125US;
+        break;
+    case MESH_AUDIO_PACKET_TIME_250US:
+        st_ptime = ST30_PTIME_250US;
+        break;
+    case MESH_AUDIO_PACKET_TIME_333US:
+        st_ptime = ST30_PTIME_333US;
+        break;
+    case MESH_AUDIO_PACKET_TIME_4MS:
+        st_ptime = ST30_PTIME_4MS;
+        break;
+    case MESH_AUDIO_PACKET_TIME_80US:
+        st_ptime = ST31_PTIME_80US;
+        break;
+    case MESH_AUDIO_PACKET_TIME_1_09MS:
+        st_ptime = ST31_PTIME_1_09MS;
+        break;
+    case MESH_AUDIO_PACKET_TIME_0_14MS:
+        st_ptime = ST31_PTIME_0_14MS;
+        break;
+    case MESH_AUDIO_PACKET_TIME_0_09MS:
+        st_ptime = ST31_PTIME_0_09MS;
+        break;
+    default:
+        return -1; // Error: unknown packet time
+    }
+
+    return 0; // Success
+}
+
+void *ST2110::get_frame_data_ptr(st_frame *src) {
+    return src->addr[0];
+}
+
+void *ST2110::get_frame_data_ptr(st30_frame *src) {
+    return src->addr;
+}
+
+int ST2110::frame_available_cb(void *ptr) {
+    auto _this = static_cast<ST2110 *>(ptr);
+    if (!_this) {
+        return -1;
+    }
+
+    _this->stop = true;
+    _this->stop.notify_all();
```

Comment:
```suggestion
    _this->frame_available = true;
    _this->frame_available.notify_one();
```

Reply:
Fixed in 133a6efcdea8895252fd1a825a2ae57a7f31cb01